pass_commander/main.py

```diff
@@ -19,149 +19,136 @@
-if os.path.isfile(tle_cache_file):
-    with open(tle_cache_file, "r") as jsonfile:
-        tle_cache = json.load(jsonfile)
-
-log.basicConfig()
-log.getLogger("apscheduler").setLevel(log.ERROR)
+import ephem
+import pydbus
+from apscheduler.schedulers.background import BackgroundScheduler
 
+from .Navigator import Navigator
+from .Radio import Radio
+from .Rotator import Rotator
+from .Station import Station
+from .Tracker import Tracker
+
+
+@dataclass
+class Config:
+    # Main
+    owmid: str
+    edl: str
+    txgain: int
+
+    # Hosts
+    radio: str
+    station: str
+    rotator: str
+
+    # Observer
+    lat: float
+    lon: float
+    alt: int
+    name: str
+
+    az_cal: int = 0
+    el_cal: int = 0
+
+    # Satellite
+    sat_id: str = "OreSat0"
+    tle_cache: dict[Union[list[str], bool]] = field(default_factory=dict)
+
+
+def load_config_file(path):
+    config_file = os.path.expanduser(path)
+    config = configparser.ConfigParser()
+    if not len(config.read(config_file)):
+        print("Config file seems to be missing. Initializing.")
+        if not os.path.exists(os.path.dirname(config_file)):
+            os.makedirs(os.path.dirname(config_file))
+        with open(config_file, "w") as f:
+            f.write(dedent("""\
+                # Be sure to replace all <hint text> including the angle brackets!
+                [Main]
+                owmid = <open weather map API key>
+                edl = <EDL command to send, hex formatted with no 0x prefix>
+                txgain = 47
+
+                [Hosts]
+                radio = 127.0.0.2
+                station = 127.0.0.1
+                rotator = 127.0.0.1
+
+                [Observer]
+                lat = <latitude in decimal notation>
+                lon = <longitude in decimal notation>
+                alt = <altitude in meters>
+                name = <station name or callsign>
+                """))
+        print(f"Please edit {config_file} before running again.")
+        sys.exit(1)
+
+    if '<' in [v[0] for s in config.keys() for v in config[s].values()]:
+        print(f"Please edit {config_file} and replace everything in <angle brackets>")
+        sys.exit(1)
+
+    # This is just shoehorned in here and ugly. Please fix!
+    def confget(conf, tree):
+        try:
+            return reduce(operator.getitem, tree, conf)
+        except KeyError:
+            print(f"Configuration element missing: {tree[0]}")
+            sys.exit(2)
+
+    return Config(
+        owmid = confget(config, ["Main", "owmid"]),
+        edl = confget(config, ["Main", "edl"]),
+        txgain = int(confget(config, ["Main", "txgain"])),
+        radio = confget(config, ["Hosts", "radio"]),
+        station = confget(config, ["Hosts", "station"]),
+        rotator = confget(config, ["Hosts", "rotator"]),
+        lat = float(confget(config, ["Observer", "lat"])),
+        lon = float(confget(config, ["Observer", "lon"])),
```

Comment:
ephem expects lat and lon to be strings. Probably because the object instantiation parser supports multiple formats in addition to decimal notation. Providing these as floats causes the ground station location to be impossible, and as a result, no sensible future passes can be calculated. Only altitude is expected to be provided as a number. Dropping these two float()'s will allow you to `python . -m all -s OreSat0.5 -a dryrun` to describe a bit about what it would do for the next 80°el minimum pass.

Reply:
This is pretty funny, by luck my testing lat/lon (45.0, -122.0) resolved to values that ephem was able to find dryrun passes for. Good catch.